core/types/transaction.go

```diff
@@ -387,6 +390,16 @@ func (tx *Transaction) RollupCostData() RollupCostData {
 	return out
 }
 
+// Conditional returns the conditional attached to the transaction
+func (tx *Transaction) Conditional() *TransactionConditional {
+	return tx.conditional
+}
+
+// SetConditional attaches a conditional to the transaction
+func (tx *Transaction) SetConditional(cond *TransactionConditional) {
+	tx.conditional = cond
+}
```

Comment:
Generally it's not advised to mutate transaction attributes. But also I don't see a better option.

It would be nice to use a generic atomic pointer util though, to make the get/set thread-safe.

Reply:
Ah yea makes sense to use an atomic pointer here, nice catch